### ISSUE TITLE
Error when calling min with key=

### ISSUE THREAD

**Issue by kumkee** [microsoft/pyright#4175]
**Describe the bug**
When calling the `min` function giving it the `key` keyword argument, pyright gives an error.
**To Reproduce**
Any attempt of calling `min` with a dictionary object and a `key` should reproduce this. See the examples given below.
**Expected behavior**
Error-free.
**Screenshots or Code**
```python
a = {1: 2, 3: 4, 5: 1}
i = min(a, key=a.get)    # this line gives an error

b = {'a': 2, 'c': 4, 'f': 1}
j = min(b, key=b.get)    # this as well
k = min(b)               # this works fine

print(i, j, k)           # 5 f a
```

The error is
```
Argument of type "Overload[(__key: int, /) -> (int | None), (__key: int, __default: int | _T@get, /) -> (int | _T@get)]" cannot
 be assigned to parameter "key" of type "(_T@min) -> SupportsRichComparison" in function "min"
      No overloaded function matches type "(int) -> SupportsRichComparison"
        Type "(__key: int, /) -> (int | None)" cannot be assigned to type "(int) -> SupportsRichComparison"
          Function return type "int | None" is incompatible with type "SupportsRichComparison"
            Type "int | None" cannot be assigned to type "SupportsRichComparison"
              Type "None" cannot be assigned to type "SupportsRichComparison"
        Type "(__key: int, __default: int | _T@get, /) -> (int | _T@get)" cannot be assigned to type "(int) -> SupportsRichCompariso
 n"
          Position-only parameter mismatch; expected 2 but received 1
          Function accepts too few positional parameters; expected 2 but received 1
```
**VS Code extension or command-line**
Neovim with Pyright 1.1.279

**Additional context**
None

**Comment by erictraut** [comment]
Pyright is doing the right thing here. Mypy gives the same error. The problem is that a.get is declared to return a value of `int | None`. You happen to know that `a.get` won't ever return a `None` in this case, but a type checker has no way of knowing this.

You can work around this by doing something like:

```python
i = min(a, key=lambda k: a.get(k, 0))
```

Or if you want to use more of a sledgehammer, a `cast` call:

```python
i = min(a, key=cast(Callable[[int], int], a.get))
```
